### ISSUE TITLE
The --test parameter leaves the skin's menu on top of the played file

### ISSUE THREAD

**Issue by pitsi** [xbmc/xbmc#24472]
### Describe the bug
When launching kodi with the --test parameter, which practically skips all menus/navigation and makes it work as a media player like mpv, the skin's menu stays on top of the playing file/stream as you can see in the screenshot below.

## Expected Behavior
Kodi should play the file/stream and show the player's menu at the bottom, and that only when the user presses m or moves the mouse.

## Actual Behavior
Instead, kodi plays the file/stream as shown in the screenshot below and the user has to press tab or the full screen button on the far right side.

## Possible Fix
No idea.

### To Reproduce
1. On a system that does NOT run kodi in standalone mode, e.g. not on an an *elec that runs it this way via its service file, launch kodi like so: `kodi --test https://test-streams.mux.dev/x36xhzz/url_6/193039199_mp4_h264_aac_hq_7.m3u8`
2. Notice the skin's main menu covering the played file/stream

### Debuglog
The debuglog can be found here:
https://paste.debian.net/hidden/ae881674/

### Screenshots 
Here are some links or screenshots to help explain the problem:
![screenshot00000](https://github.com/xbmc/xbmc/assets/5880037/6b6a4c4e-7a85-4707-ba5d-dd748bbc775b)

## Additional context or screenshots (if appropriate)
I used an internet stream above instead of a local file so that everyone can verify it plays.
Moreover, the system mentioned below is the only thing I now have that is not an *elec and thus can run kodi with user-defined parameters. This means I can not test it elsewhere.

### Your Environment
Used Operating system:
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian testing/unstable x64
 - Kodi version: 20.2

**Comment by neo1973** [comment]
I don't remember seeing this with the default estuary skin. Can you please test with that to see if the used skin makes a difference?

**Comment by pitsi** [comment]
Sure, here you go. I just did not catch the screenshot at the same frame as with confluence.

![screenshot00000](https://github.com/xbmc/xbmc/assets/5880037/52ce8006-5c74-4dcc-bab1-56c43cd479b7)

**Comment by neo1973** [comment]
Okay, I can recreate this when playing `https://test-streams.mux.dev/x36xhzz/url_6/193039199_mp4_h264_aac_hq_7.m3u8`. Interestingly this doesn't happen when I play local files, including local `.m3u8`.

**Comment by DaVukovic** [comment]
I also tested and can confirm what @neo1973 says

**Comment by sarbes** [comment]
It is just a developer feature to avoid menu navigation when testing. Why care so much about the UI state?

I mean sure, if someone wants to make it consistent, be my guest. But I don't see the issue here.

**Comment by pitsi** [comment]
You are right, it does not happen with local files, e.g. a locally stored video or a locally stored m3u with links. I should change the title. Oddly, it does not happen with remote files, e.g. this one works fine
`kodi --test http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4`

When it comes to testing something, the `--test` parameter is the closest thing I have to chromium's `--temp-profile` parameter, And that is because the `-p/--portable` parameter is completely useless in linux due to write permissions where the kodi executable is.
```
$ kodi -p
[W][06537.274385] pw.conf      | [          conf.c: 1016 try_load_conf()] can't load config client.conf: No such file or directory
[E][06537.274513] pw.conf      | [          conf.c: 1045 pw_conf_load_conf_for_context()] can't load config client.conf: No such file or directory
'this->recurse > 0' failed at ../src/pipewire/thread-loop.c:63 do_unlock()
terminate called after throwing an instance of 'spdlog::spdlog_ex'
  what():  Failed opening file /usr/share/kodi/portable_data/temp/kodi.log for writing: No such file or directory
Aborted
Crash report available at /home/user/kodi_crashlog-20240111_082954.log
```
Ignore the first lines, they are there because I do not have pipewire or pulseaudio installed.

As for the why I want something bare to test stuff, I think that the extra addons I have installed speak for themselves. In the following days I will probably open an issue on the youtube addon and on yt-dlp (for a twitch issue).
Last but not least, a few months ago I was an nvidia 340 user and I had to help on the issue kodi 20 had with that driver until 20.1 got released and fixed it. That parameter was more than helpful back then.

**Comment by mglae** [comment]
@pitsi Use
```
KODI_DATA=~/kodi_test kodi
```

**Comment by pitsi** [comment]
Omg! It worked! Thanks for the tip!

**Comment by Hitcher** [comment]
@pitsi can this issue be closed now then?

**Comment by pitsi** [comment]
My comment above was referring to the kodi_data folder, not to the issue itself.

But thanks for reminding me to test it again on kodi 21. Not only the issue is still here, it also shows no video on confluence! I had to press "full screen" to see it play. Like on 20, it does not happen for local files.

![screenshot00000](https://github.com/user-attachments/assets/43f65599-ab2f-4d0f-895d-3f999265c5c9)

![screenshot00001](https://github.com/user-attachments/assets/104063cc-08cf-4933-9c8f-949fdc7cad30)

**Comment by pitsi** [comment]
Can someone verify that this was fixed on 21.1 or 21.2? I just checked it with a local file and an online stream and it works as it should.

**Comment by pitsi** [comment]
First and foremost, I made a stupid mistake above. I had set kodi to NOT use the video as backround in the confluence skin. This is the reason that video does not show up on confluence but does on estuary.

Other than that, nothing has changed. Local and remote files play properly, but remote streams have the skin on top. Let's wait for kodi 22 if something changes.
